### ISSUE TITLE
UnnecessaryLet false positive in inner lambdas

### ISSUE THREAD

**Issue by xouabita** [detekt/detekt#3842]
When upgrading detekt to 1.7.1, I noticed some false positives for `UnnecessaryLet`.

For example:

```kotlin
fun callMe(callback: () -> Unit) {
    callback()
}
 
fun test(value: Int?) {
    value?.let { 
        callMe {
            println(it)
        }
     }
}
```

would be reported.

https://github.com/detekt/detekt/pull/3841 should fix this issue
